pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaManaged.hpp

```diff
@@ -0,0 +1,1463 @@
+    applyGeneratorDoubleExcitationMinus(const std::vector<std::size_t> &wires,
+                                        bool adjoint) {
+        static const std::string name{"GeneratorDoubleExcitationMinus"};
+        static const Precision param = 0.0;
+        const auto gate_key = std::make_pair(name, param);
+        if (!gate_cache_.gateExists(gate_key)) {
+            gate_cache_.add_gate(
+                gate_key, cuGates::getGeneratorDoubleExcitationMinus<CFP_t>());
+        }
+        applyDeviceMatrixGate(gate_cache_.get_gate_device_ptr(gate_key), {},
+                              wires, adjoint);
+        return -static_cast<PrecisionT>(0.5);
+    }
+    inline PrecisionT
+    applyGeneratorDoubleExcitationPlus(const std::vector<std::size_t> &wires,
+                                       bool adjoint) {
+        static const std::string name{"GeneratorDoubleExcitationPlus"};
+        static const Precision param = 0.0;
+        const auto gate_key = std::make_pair(name, param);
+        if (!gate_cache_.gateExists(gate_key)) {
+            gate_cache_.add_gate(
+                gate_key, cuGates::getGeneratorDoubleExcitationPlus<CFP_t>());
+        }
+        applyDeviceMatrixGate(gate_cache_.get_gate_device_ptr(gate_key), {},
+                              wires, adjoint);
+        return -static_cast<PrecisionT>(0.5);
+    }
+
+    inline PrecisionT
+    applyGeneratorMultiRZ(const std::vector<std::size_t> &wires, bool adjoint) {
+        static const std::string name{"PauliZ"};
+        static const Precision param = 0.0;
+        for (const auto &w : wires) {
+            applyDeviceMatrixGate(gate_cache_.get_gate_device_ptr(name, param),
+                                  {}, {w}, adjoint);
+        }
+        return -static_cast<PrecisionT>(0.5);
+    }
+
+    /**
+     * @brief Access the CublasCaller the object is using.
+     *
+     * @return a reference to the object's CublasCaller object.
+     */
+    auto getCublasCaller() const -> const CublasCaller & {
+        return *cublascaller_;
+    }
+
+    /**
+     * @brief Get the cuSPARSE handle that the object is using.
+     *
+     * @return cusparseHandle_t returns the cuSPARSE handle.
+     */
+    auto getCusparseHandle() const -> cusparseHandle_t {
+        if (!cusparsehandle_)
+            cusparsehandle_ = make_shared_cusparse_handle();
+        return cusparsehandle_.get();
+    }
+
+    /**
+     * @brief Get the cuStateVec handle that the object is using.
+     *
+     * @return custatevecHandle_t returns the cuStateVec handle.
+     */
+    auto getCusvHandle() const -> custatevecHandle_t { return handle_.get(); }
+
+    /**
+     * @brief Get a host data copy.
+     *
+     * @return std::vector<std::complex<PrecisionT>>
+     */
+    auto getDataVector() -> std::vector<std::complex<PrecisionT>> {
+        std::vector<std::complex<PrecisionT>> data_host(BaseType::getLength());
+        this->CopyGpuDataToHost(data_host.data(), data_host.size());
```

Comment:
`this` should be implicit here, so we can safely remove it. I think in a follow up PR we can remove the `Copy` methods from the SV class, and just use the databuffer internally.

Reply:
Sure!